core/solr_index.py

```diff
@@ -164,16 +169,16 @@ def page(self, number):
 
         # Gather facet data from the solr response
         solr_facets = solr_response.facets
+        field_counts = solr_facets.get('facet_fields')
         facets = {
-            'city': _sort_facets_asc(solr_facets, 'city'),
-            'county': _sort_facets_asc(solr_facets, 'county'),
-            'frequency': _sort_facets_asc(solr_facets, 'frequency'),
-            'language': _sort_facets_asc(solr_facets, 'language'),
-            'state': _sort_facets_asc(solr_facets, 'state'),
+            'city': _sorted_facet_counts(field_counts, 'city'),
+            'county': _sorted_facet_counts(field_counts, 'county'),
+            'frequency': _sorted_facet_counts(field_counts, 'frequency'),
+            'language': _sorted_facet_counts(field_counts, 'language'),
+            'state': _sorted_facet_counts(field_counts, 'state'),
         }
         # sort by year (desc)
-        facets['year'] = sorted(list(solr_facets['facet_ranges']['year']['counts'].items()),
-                                key = lambda k: k[0], reverse = True)
+        facets['year'] = _sorted_facet_counts(solr_facets['facet_ranges']['year'], 'counts')
```

Comment:
guess it simplifies some things not to be sorting ascending anymore!

Reply:
The sort should be the same as it was before.  It's just that the above code was originally not using a central function, which I think had something to do with solrpy magic... but now I don't recall.  In any case, I used a new function, but the new function's output should be the same as the old.  I just couldn't keep saying `_sort_facets_asc` to myself without wondering what that meant - the facets' **names** weren't sorted, after all, and they weren't sorted in any ascending manner to begin with; they're sorted by the number of hits with largest first, e.g., descending.

So it looks like I changed stuff when in fact I mostly just fixed confusing names :P

Reply:
Ah okay, I think I lost the thread on what `items()` was returning, but since it's coming off of the counts, anyway, that makes sense that it used to be sorted by counts.  That is indeed not a good name haha, thanks for fixing it!